.github/workflows/e2e-tests.yaml

```diff
@@ -0,0 +1,58 @@
+name: E2E Tests
+on:
+  workflow_dispatch:
+  push:
+    branches: [ master ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+jobs:
+  e2e-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout the source code
+      uses: actions/checkout@v4
+
+    - name: Clone e2e-tests
+      run: git clone https://github.com/AstarNetwork/e2e-tests.git --depth 1
+
+    - name: Setup node
+      uses: actions/setup-node@v4
+      with:
+        node-version: 18.x
+        cache: 'yarn'
+        cache-dependency-path: e2e-tests/yarn.lock
+
+    - name: Install deps
+      run: sudo apt -y install protobuf-compiler
+
+    - name: Install & display rust toolchain
+      run: rustup show
+
+    - name: Check targets are installed correctly
+      run: rustup target list --installed
+
+    - name: Build Astar & Shiden runtimes
+      run: cargo build --release --features astar,shiden
```

Comment:
This won't work as you expect it. We don't have _per-runtime_ feature flags implemented.

Perhaps you should use `cargo build --release -p astar-runtime -p shiden-runtime`